Sources/OpenAPIRuntime/Conversion/Converter+Server.swift

```diff
@@ -56,11 +56,34 @@ extension Converter {
                 // Drop everything after the optional semicolon (q, extensions, ...)
                 value.split(separator: ";")[0].trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
             }
-
         if acceptValues.isEmpty { return }
-        if acceptValues.contains("*/*") { return }
-        if acceptValues.contains("\(substring.split(separator: "/")[0].lowercased())/*") { return }
-        if acceptValues.contains(where: { $0.localizedCaseInsensitiveContains(substring) }) { return }
+        guard let parsedSubstring = OpenAPIMIMEType(substring) else {
+            throw RuntimeError.invalidAcceptSubstring(substring)
+        }
+        guard case .concrete(let substringType, let substringSubtype) = parsedSubstring.kind else {
+            // If the substring content type has a wildcard, just let it through.
+            // It's not well defined how such a case should behave, so be permissive.
+            return
+        }
```

Comment:
Yeah this case isn't handled correctly in this shape, I tried to explain that in the comment in the `else` block. In general we don't have clarity on what to do with e.g. `application/*` (a partial wildcard) being the _content type_ in the OpenAPI doc (as opposed to wildcards being part of the Accept header, which is well-defined). That's why I opted to go more relaxed here, as I don't know what to back up with being stricter.

I guess it comes down to whether we think that `application/*` as a content type that we have any information about whatsoever (so far I've though the answer is no, and treated it equally to `*/*`). Maybe that's not the right place to land, and we can come up with rules that constrain it more. We just run the risk of getting it slightly wrong at which point users won't have an escape hatch. But I could be convinced of going in either direction.

Reply:
I'd have thought that if we have a partial, we would validate that the top-level types match in the validate function.

I can see that a partial wildcard, e.g. `application/*`, is a strange thing to return in the `Content-Type` header field of a HTTP response, but it's clear that it doesn't match an `Accept` header of `text/plain`, or `text/*`, or anything that doesn't start with `application/`.

For this reason, I think the least surprising semantics of this validate function is to reject it.

To pull on this thread a little more though:

1. Is a partial wildcard content-type for a response valid OpenAPI (according to OAS)?
2. If yes, then is that just an oversight? I.e. if it is _valid_, is it typically always a _mistake_, or is there a genuine use case for it?

The answers to these two questions might yield some more validation and/or warning generation when the generator runs.

Reply:
OpenAPI does allow partial and full wildcards in the content type, but I've not come up with a way to treat that any differently to raw data that we know nothing about.

Agreed that we should do at least the type match. I'll update the PR.

Reply:
From chatting with folks in the team and consulting the RFCs, I think we need to adjust this differently.

For `Content-Type: a/*`, I suggested it should be permitted if `Accept: a/b` but we should only permit if `Accept: a/*` because we should treat `a/*` as a concrete MIME type with `*` being an unrecognised, but concrete (not wildcard) subtype, when in the `Content-Type` header field.

Reply:
Ok, updated with this more correct matching. Check out the unit tests, I can add more if there's a case I missed.